### PR TITLE
VACMS-6630: Move any Vet Center specific logic from va_gov_backend to va_gov_vet_center

### LINKED ISSUE #6630: Move any Vet Center specific logic from va_gov_backend to va_gov_vet_center
## User Story or Problem Statement

As our product inventory has grown we need to groom our initial catch-all module va_gov_backend.  As a code maintainer I want to confidently know where Vet Center specific code lives.

## Acceptance Criteria
- [ ] Any code in va_gov_backend that is specific to **ONLY** Vet Centers is moved to va_gov_vet_center
- [ ] Any code in that is shared among multiple products is not moved

### CMS Team

Please leave only the team that will do this work selected. If you're not sure, it's fine to leave both selected.

- [ ] `Platform CMS Team`
- [x] `Sitewide CMS Team`

### PR DESCRIPTION
**Pull request by netboss**
Closes #6630

- Ported Vet Center va_gov_vet_center_form_alter() hook and dependency functions from va_gov_vet_center.module to event subscriber.
- Ported Vet Center va_gov_vet_center JS from va_gov_backend JS.
- Updated va_gov_vet_center.libraries.yml.

## Description

Relates to #_issueid_.

## QA steps

As user _uid_ with _user_role_
1. Do this
1. Then that
1. Then validate Acceptance Criteria from issue
- [ ] This
- [ ] That
- [ ] The other thing

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Platform CMS team`
- [x] `Sitewide CMS Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Review by ethanteague — COMMENTED** [review]
Definitely on the right track here, @netboss :+1: 

You have a few phpstan notices that you will want to fix, first. 
When you see messages like `Call to an undefined method Drupal\Core\Form\FormInterface::getEntity().`  you will need to either add a "var" type comment above the line calling the method, or (preferably) wrap the method in a type check. E.g.: 
```
 if ($form_state->getFormObject() instanceof FormStateInterface) {
     $node = $form_state->getformObject()->getEntity();
}
```

File: docroot/modules/custom/va_gov_vet_center/src/EventSubscriber/EntityEventSubscriber.php [hunk 1]
```diff
@@ -73,11 +76,146 @@ public function buildNearbyVcAndOutStationFieldsetContent(FormIdAlterEvent $even
...
+   *   The form state.
+   * @param string $form_id
+   *   The form id.
+   */
+  public function optInCapHours(array &$form, FormStateInterface $form_state, $form_id) {
+    if (($form_id === 'node_vet_center_cap_edit_form') || ($form_id === 'node_vet_center_cap_form')) {
+      // We want to implement the states logic only if the field_office_hours
+      // and the field_vetcenter_cap_hours_opt_in are available.
+      /** @var \Drupal\node\NodeInterface $node*/
+      $node = $form_state->getFormObject()->getEntity();
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_vet_center/src/EventSubscriber/EntityEventSubscriber.php:160]:
```suggestion
        /** @var \Drupal\Core\Entity\EntityFormInterface $form_object */
    $form_object = $form_state->getFormObject();
    /** @var \Drupal\node\NodeInterface $node*/
    $node = $form_object->getEntity();
```
phpstan is not so good with chains of calls that each return a different type of object.  Unfortuanately you have to do it in two jumps.

Comment (netboss) [reply docroot/modules/custom/va_gov_vet_center/src/EventSubscriber/EntityEventSubscriber.php]: Thank you @swirtSJW ! Updated...

**Review by swirtSJW — COMMENTED** [review]
I was able to find matches to what was removed and what was added.  QA was a little time consuming because there was both a move (expected) and a refactor (not expected).  Fortunately the amount of code was small enough where I could still track it.  

Just waiting on passing tests before approving this.

**Comment by swirtSJW** [comment]
The tugboat instance is trying to build on master rather than main.   I am going to close this PR, then reopen after tugboat has removed the first instance.

**Comment by swirtSJW** [comment]
Looks like we have some legit behat failures

```
--- Failed scenarios:

    features/content_editing.feature:26
    features/content_editing.feature:124
    features/content_editing.feature:188
    features/save_and_continue.feature:8

134 scenarios (130 passed, 4 failed)
25m17.36s (293.15Mb)
947 steps (912 passed, 4 failed, 31 skipped)
```

Scroll up through the php log and look for red   You'l find 4 separate failures as you scroll.
https://tugboat.vfs.va.gov/log/6184ab5a690c68ad08872128

**Comment by ethanteague** [comment]
> Looks like we have some legit behat failures
> 
> ```
> --- Failed scenarios:
> 
>     features/content_editing.feature:26
>     features/content_editing.feature:124
>     features/content_editing.feature:188
>     features/save_and_continue.feature:8
> 
> 134 scenarios (130 passed, 4 failed)
> 25m17.36s (293.15Mb)
> 947 steps (912 passed, 4 failed, 31 skipped)
> ```
> 
> Scroll up through the php log and look for red You'l find 4 separate failures as you scroll. https://tugboat.vfs.va.gov/log/6184ab5a690c68ad08872128

I'll take care of this, @swirtSJW @netboss

**Comment by netboss** [comment]
@ethanteague @swirtSJW 
The missing "Save draft and continue editing" button issue seems to be in `va_gov_backend.module` line 574:
`unset($form['actions']['save_continue']);`

If I comment that line out, the "Save draft and continue editing" button shows. 

What I'm not sure is why this is unset and for which forms/content types.

Here is the whole block which mentions just the preview button:
```
    // Remove the default preview button from types not in workflow.
    /** @var \Drupal\Core\Entity\EntityFormInterface $form_object */
    $form_object = $form_state->getFormObject();
    $node_type = $form_object->getEntity()->getEntityType();

    /** @var \Drupal\workflows\WorkflowInterface $workflow */
    $workflow = \Drupal::entityTypeManager()->getStorage('workflow')->load('editorial');
    $types_in_workflow = $workflow->get('type_settings')['entity_types']['node'];
    if (!in_array($node_type, $types_in_workflow)) {
      unset($form['actions']['save_continue']);
      unset($form['actions']['preview']);
    }
```

**Comment by ethanteague** [comment]
Seeing the same accessibility error we were getting prior to adding the aria-required attribute patch to disallow checkboxes, here: https://www.drupal.org/files/issues/2021-11-05/3213473-36.patch - this pr has said patch, so a little bit confused, @netboss @swirtSJW

**Review by swirtSJW — APPROVED** [review]

**Comment by swirtSJW** [comment]
image

**Pull request merged**